### ISSUE TITLE
Request: add mangaworld.bz to sources list

### ISSUE THREAD

**Issue by Hooteo** [C9Glax/tranga#50]
Hi, 
i'm trying this docker and it seems to be very good, do you think it's possible to add this source https://www.mangaworld.bz/ to download italian manga?

Thanks

**Comment by C9Glax** [comment]
Hey there!
So far everyone has used english manga, however I have already some basic work done to also retrieve other languages (from Mangadex), so that shouldn't be too much work.

I will probably add support for different languages in the #41 release, which should be ready next week.
As for  the new connector, I will check after that release.
When everything's ready I will @ you 👍🏼

**Comment by C9Glax** [comment]
@Hooteo if you want you can try the cuttingedge branch.
It includes both the language select for mangadex as well as the new connector.

Feedback is appreciated!

**Comment by Hooteo** [comment]
I'm sorry i don't get it. How can i set the different language? I have update my docker compose but i don't have this possibility.

Thanks

**Comment by C9Glax** [comment]
Have you changed both tranga and website to `cuttingedge`?
If so maybe clear your browser cache or hard refresh

**Comment by Hooteo** [comment]
This ismy docker-compose yaml:

version: '3'
services:
  tranga-api:
    image: glax/tranga-api:latest
    container_name: tranga-api
    volumes:
      - /opt/cont/tranga:/usr/share/Tranga-API
      - /dati/kavita/manga:/Manga
    ports:
      - "6531:6531"
    restart: unless-stopped
  tranga-website:
    image: glax/tranga-website:latest
    container_name: tranga-website
    volumes:
      - /opt/cont/tranga/imageCache:/usr/share/nginx/html/imageCache:ro
    ports:
      - "9555:80"
    depends_on: 
      - tranga-api
    restart: unless-stopped

I tried to update to the new version but i don't have any possoblity to change the language.

![image](https://github.com/C9Glax/tranga/assets/17424781/da3fce5c-6c8d-464e-9864-4d4255dbb6b4)

Only english:

![image](https://github.com/C9Glax/tranga/assets/17424781/2779f301-ed03-4c51-941e-137d479e7c24)

**Comment by C9Glax** [comment]
`image: glax/tranga-api:latest` is your problem.

Use the `cuttingedge` branch version instead:
https://github.com/C9Glax/tranga/blob/cuttingedge/docker-compose.yaml

**Comment by Hooteo** [comment]
@C9Glax  it's working perfectly, i think you did a perfect job.

![image](https://github.com/C9Glax/tranga/assets/17424781/8efd39de-680c-4751-adc4-2bcee5ce1c12)

Only MangaDex can use different language? 

Thanks

**Comment by C9Glax** [comment]
Thanks!

Mangadexx is the only site supporting multiple languages.
Mangasee, Manganato, Mangakatana are all english only
Mangaworld italian only.

**Comment by Hooteo** [comment]
@C9Glax  ok now i found a problem with Mangaworld, some chapters needs the login to be downloaded (you pay a lifetime account to download everything) . Do you think it's possible to do the login and implement it in the settings? 

MangaDex is the only one that support other languages but it has the problem to block your ip...

**Comment by C9Glax** [comment]
Hmm, I haven't done anything of that sort yet... So maybe at some point...

MangaDex only blocks your IP (as far as I know) if you exceed rate-limits, tranga doesn't do that, so that shouldn't be a problem

**Comment by C9Glax** [comment]
Is this what you are talking about?
![grafik](https://github.com/C9Glax/tranga/assets/13404778/4effbb07-55f3-4ebe-a495-fa4deb15b3ef)

Because if so, there probably won't be a problem... I am scraping the webpage images, so if you can see the website with the chapter-images, you should be able to download it...

But if you say, that there was an error, could you maybe send the log-output?
`docker logs tranga-api`

**Comment by Hooteo** [comment]
Manga web i have a lot of these errors:

2023/09/14 12:05:43 [error] 30#30: *10 open() "/usr/share/nginx/html/media/cover.jpg" failed (2: No such file or directory), client: 192.168.69.128, server: localhost, request: "GET /media/cover.jpg HTTP/1.1", host: "192.168.69.4:9555", referrer: "http://192.168.69.4:9555/"
192.168.69.128 - - [14/Sep/2023:12:05:43 +0000] "GET /media/cover.jpg HTTP/1.1" 404 555 "http://192.168.69.4:9555/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.76" "-"
192.168.69.128 - - [14/Sep/2023:12:05:44 +0000] "GET / HTTP/1.1" 304 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.76" "-"
2023/09/14 12:05:44 [error] 30#30: *10 open() "/usr/share/nginx/html/media/cover.jpg" failed (2: No such file or directory), client: 192.168.69.128, server: localhost, request: "GET /media/cover.jpg HTTP/1.1", host: "192.168.69.4:9555", referrer: "http://192.168.69.4:9555/"
192.168.69.128 - - [14/Sep/2023:12:05:44 +0000] "GET /media/cover.jpg HTTP/1.1" 404 555 "http://192.168.69.4:9555/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.76" "-"
192.168.69.128 - - [14/Sep/2023:12:38:11 +0000] "GET / HTTP/1.1" 304 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.76" "-"
192.168.69.128 - - [14/Sep/2023:12:38:11 +0000] "GET /media/cover.jpg HTTP/1.1" 404 555 "http://192.168.69.4:9555/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.76" "-"
2023/09/14 12:38:11 [error] 30#30: *11 open() "/usr/share/nginx/html/media/cover.jpg" failed (2: No such file or directory), client: 192.168.69.128, server: localhost, request: "GET /media/cover.jpg HTTP/1.1", host: "192.168.69.4:9555", referrer: "http://192.168.69.4:9555/"

For manga api i don't have any logs:

![image](https://github.com/C9Glax/tranga/assets/17424781/7ec7440e-f389-420d-826e-db5890426350)

![image](https://github.com/C9Glax/tranga/assets/17424781/ca77bc58-a7f1-4064-bded-46fff12643e8)

When i try do download jujutsu kaisen it only downloaded Vol 24 Ch 01 all the other chapters are stuck:

![image](https://github.com/C9Glax/tranga/assets/17424781/cecf5896-e26c-4fac-a062-583e70911ecf)

**Comment by C9Glax** [comment]
Okay the first part doesn't matter, that's just a missing cover file that has been missing forever...
Then again I am not the smartes tool in the shed, and I forgot to add the logger to tranga-api...
Could you update `docker compose pull && docker compose up -d` for me, and then post the output of `docker logs tranga-api` again?

**Comment by Hooteo** [comment]
Yes now i have the logs: [](https://we.tl/t-Rkzs1su8pv) .
So from what i notice it downloaded vol 24 chapter 02 and then it got stuck:

[09/14/2023 13:13:31.399]    Mangaworld | Downloading image 019/026
[09/14/2023 13:13:32.480]    Mangaworld | Downloading image 020/026
[09/14/2023 13:13:33.572]    Mangaworld | Downloading image 021/026
[09/14/2023 13:13:34.676]    Mangaworld | Downloading image 022/026
[09/14/2023 13:13:35.748]    Mangaworld | Downloading image 023/026
[09/14/2023 13:13:36.859]    Mangaworld | Downloading image 024/026
[09/14/2023 13:13:37.931]    Mangaworld | Downloading image 025/026
[09/14/2023 13:13:39.16 ]    Mangaworld | Downloading image 026/026
[09/14/2023 13:13:40.130]    Mangaworld | Creating archive /Manga/Jujutsu Kaisen/Jujutsu Kaisen - Vol.24 Ch.02 .cbz
[09/14/2023 13:13:41.37 ]       JobBoss | Removing DownloadChapter Tranga.Jobs.DownloadChapter-anVqdXRzdWthaXNlbjIwMTg=-02 Chapter Jujutsu Kaisen anVqdXRzdWthaXNlbjIwMTg= 02 
[09/14/2023 13:13:41.37 ] DownloadChapt | Cancelling DownloadChapter Tranga.Jobs.DownloadChapter-anVqdXRzdWthaXNlbjIwMTg=-02 Chapter Jujutsu Kaisen anVqdXRzdWthaXNlbjIwMTg= 02 
[09/14/2023 13:13:41.38 ]       JobBoss | Removing 0 jobs.
[09/14/2023 13:13:41.38 ]       JobBoss | Exporting Jobs
[09/14/2023 13:13:41.149]    Mangaworld | Retrieving chapter-info Chapter Jujutsu Kaisen anVqdXRzdWthaXNlbjIwMTg= 02  Publication Jujutsu Kaisen anVqdXRzdWthaXNlbjIwMTg=
[09/14/2023 13:13:41.348]    Mangaworld | Downloading Images for /Manga/Jujutsu Kaisen/Jujutsu Kaisen - Vol.24 Ch.02 .cbz

In fact no chapter is taken for download:

![image](https://github.com/C9Glax/tranga/assets/17424781/4b2f98ce-9efd-4c04-b5e1-4551a6a929f0)

If i try to search solo leveling it doesn't show me anything:

![image](https://github.com/C9Glax/tranga/assets/17424781/e9fbf587-42a1-4a2c-a1ea-ce5f20955cc4)

Probably the problem is that the cover downloaded for solo leveling is in .png instead for the other manga is jpg:

Solo leveling logs:

[09/14/2023 13:18:48.974]    Mangaworld | Searching Publications. Term="solo"
[09/14/2023 13:18:51.769]    **Mangaworld | Saving cover to /var/lib/tranga-api/imageCache/5fab3399aadcf571527080e7.png**

Jujutsu logs:

[09/14/2023 13:12:58.245]    Mangaworld | Searching Publications. Term="jujutsu"
[09/14/2023 13:13:00.943]    Mangaworld | Saving cover to /var/lib/tranga-api/imageCache/5f9f94492041561e75ffff50.jpg
[09/14/2023 13:13:03.200]    Mangaworld | Saving cover to /var/lib/tranga-api/imageCache/5f9f989bf6abbe1e93b023d3.jpg

**Comment by C9Glax** [comment]
Sorry for getting back so late.
Sidenote: If you can in the furture put logs in ```these blocks``` (it's 3 ` on both sides) that would increase readability.

> `[09/14/2023 13:13:41.348] Mangaworld | Downloading Images for /Manga/Jujutsu Kaisen/Jujutsu Kaisen - Vol.24 Ch.02 .cbz` Indicates that the task is started, but nothing happening probably means that an error occured, but wasn't logged.
Also Vol.24 Ch.02 doesn't exist, but that should be an easy fix.

> Probably the problem is that the cover downloaded for solo leveling is in .png instead for the other manga is jpg

Shouldn't really matter, but I will look into it.

**Comment by C9Glax** [comment]
Okay I just pushed to cuttingedge.

You couldn't find `jujutsu kaisen`, because i made a formatting mistake on titles with spaces, returning no results.
Adding via url still worked fine, because that query was never made in that case.

HTML-Agilitypack apparently works in mysterious ways with xpath, because it doesn't care what node you execute xpath on, it always takes the documentElement as root, returning the same element, even if you were knee deep down the tree.
**In short:** volume and chapter numbers were messed up, causing jobs to not work properly.

**Comment by Hooteo** [comment]
It sounds good. Now I'm in vacation for a week and I don't have access to my home lab. Once I'll be back I'll update you. Thanks

**Comment by Hooteo** [comment]
Hi @C9Glax i tried the new version and it's working fine. It's possible to schedule the check for new characters and download them automatically?

**Comment by C9Glax** [comment]
Copy & Paste from README:

> This project downloads the images for a Manga from the specified Scanlation-Website and packages them with some metadata - from that same website - in a .cbz-archive (per chapter).
**It does this on an interval**, and checks for any Chapters (.cbz-Archive) not already existing in your specified Download-Location. (If you rename or move files, it will download those again)

So it already does that.

Good to hear that it works though!

**Comment by bubez81** [comment]
Hello, with mangaworld I receive this error:

[01/18/2024 00:29:31.582]       JobBoss | Next job in 00:00:00 Tranga.Jobs.UpdateMetadata-anVqdXRzdWthaXNlbjIwMTg=
[01/18/2024 00:29:31.683] HttpDownloadC | Waiting 0.7795578 seconds
[01/18/2024 00:29:32.617] HttpDownloadC | Request-Error NotFound: Not Found
[01/18/2024 00:29:32.617] UpdateMetadat | Could not find Manga Publication Jujutsu Kaisen anVqdXRzdWthaXNlbjIwMTg=
[01/18/2024 00:29:32.718]       JobBoss | Removing Tranga.Jobs.UpdateMetadata-anVqdXRzdWthaXNlbjIwMTg= Manga: Publication Jujutsu Kaisen anVqdXRzdWthaXNlbjIwMTg=
[01/18/2024 00:29:32.718] UpdateMetadat | Cancelling Tranga.Jobs.UpdateMetadata-anVqdXRzdWthaXNlbjIwMTg= Manga: Publication Jujutsu Kaisen anVqdXRzdWthaXNlbjIwMTg=
[01/18/2024 00:29:32.718]       JobBoss | Deleting Job-file /usr/share/tranga-api/jobs/Tranga.Jobs.UpdateMetadata-anVqdXRzdWthaXNlbjIwMTg=.json
[01/18/2024 00:29:32.719]       JobBoss | Next job in 02:59:58.8630081 Tranga.Jobs.DownloadNewChapters-anVqdXRzdWthaXNlbjIwMTg=
[01/18/2024 00:30:09.75 ]        Kavita | Creating libraryConnector Kavita
[01/18/2024 00:30:09.75 ]        Server | Adding Kavita https://kavita.perlhof.dyndns.org
[01/18/2024 00:30:09.75 ]        Server | Exporting libraryConnectors
[01/18/2024 00:30:32.699]        Kavita | Creating libraryConnector Kavita
[01/18/2024 00:30:32.699]        Server | Adding Kavita https://kavita.perlhof.dyndns.org
[01/18/2024 00:30:32.700]        Server | Exporting libraryConnectors
[01/18/2024 00:30:45.312]       JobBoss | Added Tranga.Jobs.UpdateMetadata-anVqdXRzdWthaXNlbjIwMTg= Manga: Publication Jujutsu Kaisen anVqdXRzdWthaXNlbjIwMTg=

did I do something wrong?
